lib/pure/httpclient.nim

```diff
@@ -447,34 +478,33 @@ proc generateHeaders(requestUrl: Uri, httpMethod: string,
     result.add($modifiedUrl)
 
   # HTTP/1.1\c\l
-  result.add(" HTTP/1.1\c\L")
+  result.add(" HTTP/1.1" & cl)
 
   # Host header.
   if not headers.hasKey("Host"):
     if requestUrl.port == "":
-      add(result, "Host: " & requestUrl.hostname & "\c\L")
+      add(result, "Host: " & requestUrl.hostname & cl)
     else:
-      add(result, "Host: " & requestUrl.hostname & ":" & requestUrl.port & "\c\L")
+      add(result, "Host: " & requestUrl.hostname & ":" & requestUrl.port & cl)
 
   # Connection header.
   if not headers.hasKey("Connection"):
-    add(result, "Connection: Keep-Alive\c\L")
+    add(result, "Connection: Keep-Alive" & cl)
 
   # Content length header.
   const requiresBody = ["POST", "PUT", "PATCH"]
-  let needsContentLength = body.len > 0 or upperMethod in requiresBody
-  if needsContentLength and not headers.hasKey("Content-Length"):
-    add(result, "Content-Length: " & $body.len & "\c\L")
+  if upperMethod in requiresBody and not headers.hasKey("Content-Length"):
+    add(result, "Content-Length: 0" & cl)
```

Comment:
You're not adding a `Content-Length` here when there is a body supplied as far as I can see? If you're adding this header somewhere else then it introduces risks for bugs, can we move this logic here please?